File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);
int grade(int result_of_letters, int result_of_words, int result_of_sentences);

int main(void)
{
    string text = get_string("Text: ");
    int result_of_letters = 0, result_of_words = 0, result_of_sentences = 0;

    result_of_letters = count_letters(text);
    result_of_words = count_words(text);
    result_of_sentences = count_sentences(text);

    int index = grade(result_of_letters, result_of_words, result_of_sentences);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (index > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }
    return 0;
}

int count_letters(string text)
{
    int length = strlen(text);
    int count = 0;

    for (int i = 0; i < length; i++)
        if (isalpha(text[i]))
        {
            count++;
        }

    return count;
}

int count_words(string text)
{
    int length = strlen(text);
    int count = 0;

    for (int j = 0; j < length; j++)
        if (isspace(text[j]))
        {
            count++;

        }

    return count + 1;
}

int count_sentences(string text)
{
    int length = strlen(text);
    int count = 0;
    int k;

    for (k = 0; k < length; k++)
    {
        if (text[k] == '.' || text[k] == '!' || text[k] == '?')
        {
            count++;
        }
    }
    return count;
}

int grade(int result_of_letters, int result_of_words, int result_of_sentences)
{
    float L, S, index;
    L = (float) result_of_letters / (float)result_of_words * 100;
    S = (float) result_of_sentences / (float)result_of_words * 100;

    index = round((0.0588 * L) - (0.296 * S) - (15.8));
    int grade = (index);

    return grade;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ").lower()

    result_of_letters = count_letters(text)
    result_of_words = count_words(text)
    result_of_sentences = count_sentences(text)
    index = grade(result_of_letters, result_of_words, result_of_sentences)

    if index < 1:
        print("Before Grade 1")

    elif index > 16:
        print("Grade 16+")

    else:
        print("Grade ", index)

    return 0
    

def count_letters(text):
    length = len(text)
    count = 0

    for i in range(length):
        if text[i].isalpha() == True:
            count += 1
    return count


def count_words(text):
    length = len(text)
    count = 0

    for j in range(length):
        if text[j].isspace() == True:
            count += 1
    return count + 1


def count_sentences(text):
    length = len(text)
    count = 0

    for k in range(length):
        if text[k] in [".", "!", "?"]:
            count += 1
    print(count)
    return count


def grade(result_of_letters, result_of_words, result_of_sentences):
    L = (result_of_letters * 100) / result_of_words
    S = (result_of_sentences * 100) / result_of_words

    index = round(0.0588 * L - 0.296 * S - 15.8)
    grade = index

    return grade


main()